app/code/Magento/CatalogUrlRewrite/Model/Products/AdaptUrlRewritesToVisibilityAttribute.php

```diff
@@ -18,7 +18,9 @@
 use Magento\UrlRewrite\Service\V1\Data\UrlRewrite;
 
 /**
- *  Save/Delete UrlRewrites by Product ID's and visibility
+ * Save/Delete UrlRewrites by Product ID's and visibility
+ *
+ * @deprecated
```

Comment:
Could you write some info why it was deprecated and what is/should be used instead? Such info will be really helpful during investigations